wsd/DocumentBroker.cpp

```diff
@@ -1823,16 +1859,18 @@ bool DocumentBroker::autoSave(const bool force, const bool dontSaveIfUnmodified)
 
     LOG_TRC("autoSave(): forceful? " << force
                                      << ", dontSaveIfUnmodified: " << dontSaveIfUnmodified);
-    if (_sessions.empty() || _storage == nullptr || !isLoaded() || (!isModified() && !force))
+
+    const CanSave canSave = canSaveToDisk();
+    if (canSave != CanSave::Yes)
     {
-        // Nothing to do.
-        LOG_TRC("Nothing to autosave [" << _docKey << "].");
+        LOG_DBG("Cannot save to disk: " << name(canSave));
         return false;
     }
 
-    if (_docState.isDisconnected())
```

Comment:
Being disconnected from the Kit is handled by `canSaveToDisk()` above. We only need to check we are modified, or forced autosave, to proceed.